### PR TITLE
AWS-S3 step

### PR DESCRIPTION
**Pull request by mnmlluke**
Created step to upload any file or directory to S3 using the Storage Integration.

### CODE REVIEW

File: incubating/aws-s3/src/start.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export AWS_SECRET_ACCESS_KEY=$(codefresh get context ${S3_INTEGRATION} --decrypt -o json |jq --raw-output '.spec.data.auth.jsonConfig.secretAccessKey')
+export AWS_DEFAULT_REGION=$(echo "${REGION,,}")
+export AWS_PAGER=""
+
+cd $working_directory
+echo "Starting to upload files to S3"
+echo ""
+
+if [ -d $SOURCE ]; then
+  aws s3 cp $SOURCE s3://$BUCKET/$S3_PREFIX --recursive
```
Comment (lrochette) [inline incubating/aws-s3/src/start.sh:11]: Should we test for some error?

File: incubating/aws-s3/step.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  maintainers:
+    - name: Luke Goodfellow
+    - email: luke.goodfellow@codefresh.io
+  categories:
+    - utilities
+  official: true
+  tags: []
+  icon:
+    type: svg
+    url: https://raw.githubusercontent.com/codefresh-io/steps/aws-s3/incubating/aws-cdk/images/s3-logo.svg
```
Comment (lrochette) [inline incubating/aws-s3/step.yaml:20]: I think you're missing the branch and the aws-s3 should be below incubating
For reference, the one for ServiceNow plugin
https://raw.githubusercontent.com/codefresh-io/steps/master/incubating/service-now/images/servicenow-2.png

so  
https://raw.githubusercontent.com/codefresh-io/steps/master/incubating/aws-s3/images/s3-logo.svg

File: incubating/aws-s3/step.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+              "default": "us-east-1"
+            },
+          "BUCKET": {
+              "type": "string",
+              "description": "The name of the bucket you are using"
+            },
+          "S3_PREFIX":
+            {
+              "type": "string",
+              "description": "The directory in S3 you want the files to be uploaded to"
```
Comment (lrochette) [inline incubating/aws-s3/step.yaml:72]: if the name does not end by '/' that will be the name of the file (for a single file upload)
Maybe put some kind of note

File: incubating/aws-s3/src/start.sh (hunk 1 above)
Comment (mnmlluke) [reply incubating/aws-s3/src/start.sh:11]: The AWS command will error out when the directory is missing

File: incubating/aws-s3/step.yaml (hunk 2 above)
Comment (mnmlluke) [reply incubating/aws-s3/step.yaml]: You are Correct I will make the correction.

File: incubating/aws-s3/step.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            },
+          "SOURCE":
+            {
+              "type": "string",
+              "description": "The directory or name of file you want to upload to S3. Path from /codefresh/volume/."
```
[the remark this answers is not in the document]
Comment (mnmlluke) [reply incubating/aws-s3/step.yaml]: Updated Wording

File: incubating/aws-s3/step.yaml (hunk 3 above)
Comment (mnmlluke) [reply incubating/aws-s3/step.yaml]: Updated the wording

File: incubating/aws-s3/src/start.sh (hunk 1 above)
Comment (mnmlluke) [reply incubating/aws-s3/src/start.sh:11]: The only thing is that I have seen when testing, it wont cause the step to fail. Not sure how to fix that.

File: incubating/aws-s3/step.yaml [hunk 5]
```diff
@@ -17,7 +17,7 @@ metadata:
   tags: []
   icon:
     type: svg
-    url: https://raw.githubusercontent.com/codefresh-io/steps/aws-s3/incubating/aws-cdk/images/s3-logo.svg
+    url: https://raw.githubusercontent.com/codefresh-io/steps/master/aws-s3/incubating/aws-cdk/images/s3-logo.svg
```
Comment (lrochette) [inline incubating/aws-s3/step.yaml:20]: this is still incorrect, still a reference to aws-cdk

**Review by lrochette — CHANGES_REQUESTED** [review]

File: incubating/aws-s3/step.yaml (hunk 5 above)
Comment (mnmlluke) [reply incubating/aws-s3/step.yaml:20]: Updated URL

**Review by lrochette — APPROVED** [review]

**Pull request merged**